src/app/store/slices/storage/storage.thunks/downloadItemsThunk.ts

```diff
@@ -1,70 +1,134 @@
-import { DriveFileData, DriveFolderData, DriveItemData } from 'app/drive/types';
 import i18n from 'app/i18n/services/i18n.service';
 import notificationsService, { ToastType } from 'app/notifications/services/notifications.service';
-import { DownloadFileTask, DownloadFolderTask, TaskType } from 'app/tasks/types';
+import date from 'app/core/services/date.service';
+import { FlatFolderZip } from 'app/core/services/zip.service';
+import { downloadFile } from 'app/network/download';
+import { DriveItemData } from 'app/drive/types';
+import localStorageService from 'app/core/services/local-storage.service';
 import tasksService from 'app/tasks/services/tasks.service';
+import { DownloadFileTask, TaskStatus, TaskType } from 'app/tasks/types';
+import folderService from 'app/drive/services/folder.service';
+import errorService from 'app/core/services/error.service';
 
 export const downloadItemsThunk = createAsyncThunk<void, DriveItemData[], { state: RootState }>(
   'storage/downloadItems',
-  async (items: DriveItemData[], { dispatch, requestId, rejectWithValue }) => {
+  async (items: DriveItemData[], { rejectWithValue }) => {
     const errors: unknown[] = [];
-    const taskGroupId = requestId;
-    const tasksIds: string[] = [];
-
-    // * 1. Creates tasks
-    for (const item of items) {
-      if (item.isFolder) {
-        const taskId = tasksService.create<DownloadFolderTask>({
-          action: TaskType.DownloadFolder,
-          relatedTaskId: taskGroupId,
-          folder: item,
-          compressionFormat: 'zip',
-          showNotification: true,
-          cancellable: true,
-        });
+    const downloadProgress: number[] = [];
+    const abortController = new AbortController();
+    const formattedDate = date.format(new Date(), 'DD/MM/YYYY - HH:mm');
+    const folderName = `Internxt (${formattedDate})`;
+    const folder = new FlatFolderZip(folderName, {});
 
-        tasksIds.push(taskId);
-      } else {
-        const taskId = tasksService.create<DownloadFileTask>({
-          action: TaskType.DownloadFile,
-          file: item,
-          showNotification: true,
-          cancellable: true,
-          relatedTaskId: taskGroupId,
-        });
+    const user = localStorageService.getUser();
+    if (!user) throw new Error('User not found');
 
-        tasksIds.push(taskId);
-      }
-    }
+    const taskId = tasksService.create<DownloadFileTask>({
+      action: TaskType.DownloadFile,
 
-    // * 2. Executes tasks
-    for (const [index, item] of items.entries()) {
-      const taskId = tasksIds[index];
-
-      if (item.isFolder) {
-        await dispatch(
-          storageThunks.downloadFolderThunk({
-            folder: item as DriveFolderData,
-            options: { taskId },
-          }),
-        );
-      } else {
-        await dispatch(
-          storageThunks.downloadFileThunk({
-            file: item as DriveFileData,
-            options: { taskId },
-          }),
-        );
```

Comment:
Need to restore this because now is downloading every item as a zip

Reply:
restored here: https://github.com/internxt/drive-web/pull/575/commits/152f6d3acbe854a2cc76dd35905e1c5edf1d16e6